docs/DeveloperGuide.md

```diff
@@ -1,38 +1,392 @@
+
+Step 2. The `MarkCommand` is then executed by the `LogicManager` calling `execute()`. Then, the  `ModuleList`, which is retrieved from the `ModelManager` through `getModuleList()`, is iterated through to find the index of a `Module` with the same corresponding `moduleCode` through `getModule().getCode()`. This index is assigned to a variable.
+
+Step 3. It is then passed to the `ModelManager`, along with the index and grade, and executes the `markModule()` method. It retrieves the `Module` object from the `ModuleList` through the `getModule(index)` method. The `markTaken()` method is then called on this `Module` which sets the `isTaken` attribute as true and saves the grade. 
+
+Step 4. Upon successful execution of the above, it is then passed back to `MarkCommand` where  `CommandResult()` is constructed with the message to be printed to the user.
+
+Step 5. The `CommandResult` object is passed to the `Ui` component with a `printMessage()` method which prints the formatted message to the Command Line Interface.
+
+The following sequence diagram shows how the `mark` command works:
+![MarkModuleSequenceDiagram](uml/diagrams/MarkModSequence.png)
+
+<br>
+
+### List modules
+The List modules feature allows users to view their added modules, in a specified range using the command `list`. There are 3 ways of modules listing :
+  1. List all modules in the planner
+  2. List all modules in the planner for a specific year
+  3. List all modules in the planner for a specific year and semester
+
+Given below is an example usage scenario for each type, and how the list modules mechanism behaves at each step.
+**When a the year and semester are not specified:**
+Step 1. The user executes the command `list`, without any specified year or semester range, to list all modules in the planner. The `list` command is executed within the switch case of the `parseCommand()` method of `CommandParser`. 
+
+Step 2. If a valid command is entered, the `printModule` method of `ModuleList` will be executed with the inputs of -1 for both the year and semester. 
+
+Step 3. In `printModule` of `ModuleList`, a `Map<Integer, Map<Integer, List<String[]>>>` Hashmap is iniitialised, and all the modules stored in the `modules` container is added to the Hashmap. 
+
+Step 4. The year and semester have values of -1, which then `printModules` recognises as printing all modules in the Hashmap. A `List<String[]> modules` is initialised with all the modules in the Hashmap. 
+
+Step 5. If `modules` is not empty, the modules for that year and semester are printed sequentially. For modules with available grade information, the grade will be printed beside the module code. 
+
+Below is a sequence diagram of how the `list` command works:
+![ListSequenceDiagram](uml/diagrams/ListSequence.png)
+
```

Comment:
I found there are many parts are the same across all the sequential diagram, I'm thinking if there is any way to simply these part so that the different or important part can be emphasized.